src/hera/workflows/v5/script.py

```diff
@@ -126,12 +173,12 @@ def _build_template(self) -> _ModelScriptTemplate:
             image_pull_policy=self._build_image_pull_policy(),
             lifecycle=self.lifecycle,
             liveness_probe=self.liveness_probe,
-            name=self.name,
```

Comment:
Is this necessary because of the clash with the template name field, which is required?

Reply:
Precisely!

Reply:
Small: what do you think about setting `template_name` on the template mixin rather than `container_name` on the subclass? I _think_ that might be easier for users since there's a single name field to set on the template. By comparison, having `container_name` does not fully align with the concept of a task/step 🤔

Reply:
gotcha - I don’t think the script name is important/actually used in any of the examples. The reason I called it container_name was because of 
> name	string	Name of the container specified as a DNS_LABEL. Each container in a pod must have a unique name (DNS_LABEL). Cannot be updated.

from https://argoproj.github.io/argo-workflows/fields/#scripttemplate

Given that the `name` field users generally want to set is the template name, I wanted to keep that short/simple.

Reply:
Got it! What do you think about making the name the template name if the `container_name` is not specified? 🤔